snapcraft/storeapi/_upload.py

```diff
@@ -125,9 +161,8 @@ def upload_files(binary_filename, config=None):
             'An unexpected error was found while uploading files.')
         result['errors'] = [str(err)]
     finally:
-        # make sure to close any open files used for request
-        for fd in files.values():
-            fd.close()
+        # Close the open file
+        binary_file.close()
```

Comment:
Right, this may need to be refactored to support multiple snap uploads when library snaps come (hope we can have something for next week).